cost/flexera/cco/spend_percentage_change/spend_percentage_change.pt

```diff
@@ -0,0 +1,506 @@
+    host rs_optima_host
+    path join(["/bill-analysis/orgs/", rs_org_id, "/costs/dimensions"])
+    header "Api-Version", "1.0"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "dimensions[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "name", jmes_path(col_item, "name")
+      field "type", jmes_path(col_item, "type")
+    end
+  end
+end
+
+datasource "ds_dimension" do
+  run_script $js_dimension, $ds_cost_dimensions, $param_dimension
+end
+
+script "js_dimension", type: "javascript" do
+  parameters "ds_cost_dimensions", "param_dimension"
+  result "result"
+  code <<-EOS
+  result = null
+
+  if (param_dimension != "") {
+    found_id = _.find(ds_cost_dimensions, function(item) {
+      return item['id'].toLowerCase().trim() == param_dimension.toLowerCase().trim()
+    })
+
+    found_name = _.find(ds_cost_dimensions, function(item) {
+      return item['name'].toLowerCase().trim() == param_dimension.toLowerCase().trim()
+    })
+
+    if (param_dimension.toLowerCase() == 'billing center') {
+      result = { name: 'Billing Center', id: 'billing_center_id' }
+    } else if (found_id != undefined) {
+      result = found_id
+    } else if (found_name != undefined) {
+      result = found_name
+    }
+  }
+EOS
+end
+
+datasource "ds_raw_spend" do
+  request do
+    run_script $js_raw_spend, $ds_billing_centers, $ds_top_level_bcs, $ds_dimension, $param_months, $param_cost_metric, $param_dimension_filters, rs_org_id, rs_optima_host
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "rows[*]") do
+      field "metrics", jmes_path(col_item, "metrics")
+      field "dimensions", jmes_path(col_item, "dimensions")
+      field "timestamp", jmes_path(col_item, "timestamp")
+    end
+  end
+end
+
+script "js_raw_spend", type: "javascript" do
+  parameters "ds_billing_centers", "ds_top_level_bcs", "ds_dimension", "param_months", "param_cost_metric", "param_dimension_filters", "rs_org_id", "rs_optima_host"
+  result "request"
+  code <<-EOS
+  cost_metric_table = {
+    "Unamortized Unblended": "cost_nonamortized_unblended_adj",
+    "Amortized Unblended": "cost_amortized_unblended_adj",
+    "Unamortized Blended": "cost_nonamortized_blended_adj",
+    "Amortized Blended": "cost_amortized_blended_adj"
+  }
+
+  billing_center_table = {}
+  _.each(ds_billing_centers, function(bc) { billing_center_table[bc['name']] == bc['id'] })
+
+  billing_center_ids = ds_top_level_bcs
+  dimensions = null
+  filter = null
+
+  if (ds_dimension != null) {
+    dimensions = [ ds_dimension['id'] ]
+
+    if (ds_dimension['id'] == 'billing_center_id' && param_dimension_filters.length > 0) {
+      billing_center_ids = _.map(param_dimension_filters, function(item) { return billing_center_table[item] })
+    } else if (param_dimension_filters.length > 0) {
+      filters = _.map(param_dimension_filters, function(item) {
+        return { "dimension": ds_dimension['id'], "type": "equal", "value": item }
```

Comment:
Thinking this would be list param accepting key==value pairs (amongst other) like we have filters implemented [here](https://github.com/flexera-public/policy_templates/blob/580d517139729902fefa05acedc67f39ab5d8441/cost/aws/superseded_ebs_volumes/aws_superseded_ebs_volumes.pt#L495-L560)

So I could notify portfolio owner with alert on % change increase for any "App Code" under Portfolio==Customer Service using PT with params:
`param_dimension_filters=["Portfolio==Customer Service"]` (filter on Customer Service costs)
`param_dimension=App Code` (group costs by app code. There are multiple app codes under a portfolio)
`param_email=["CustomerServicePortfolioOwner@contoso.com"]` (notify Portfolio Owner)

Happy to propose a suggested PT change as well if helpful